### PR TITLE
intital commit

### PR DESCRIPTION
**Pull request by efyudi**

### CODE REVIEW

File: fakhar-weatherman.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+        """ Function to Print calculated readings (Debugging Purposes) """
+        day = weather_readings[recorded_month][entry_in_file].day
+        month = int(weather_readings[recorded_month][entry_in_file].month)
+        print(whatToFindDictionary[index] + " : " + str(value) + unitsDictionary[index] +
+              " on " + monthDictionary[month] + " " + day)
+
+    @staticmethod
+    def check_max(old_max, value):
+        """ Find Max """
+        old_max = value if value > old_max else old_max
```
Comment (ammadanwer) [inline fakhar-weatherman.py]: `return value if value > old_max else old_max`

File: fakhar-weatherman.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+        """ Function will find MAX of what is specified, based on the index passed """
+        find_max = -sys.maxsize
+        highest_recorded_month = 0
+        entry_in_file_max = 0
+        ''' month here is, number of the MONTH FILE. ex : files read are 2. 0 = AUG, 1 = OCT.
+         so I decode it later using the month variable '''
+        for month in range(0, len(
+                weather_readings)):
+            for entry in range(0, len(weather_readings[month])):
+                if weather_readings[month][entry].attributes[index] == "":  # if empty, skips
```
Comment (ammadanwer) [inline fakhar-weatherman.py]: move the comment above this line

File: fakhar-weatherman.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+                min_value += int(weather_readings[month]
+                                 [reading].attributes[3]) if weather_readings[month][reading].attributes[3] else 0
+        return max_value / int(total_readings), min_value / int(total_readings)
+
+    def calculate_avg(self, weather_readings):
+        """Calculates the Average Temperatures and Humidity"""
+        avgmax_temp, avgmin_temp = self.calculate_mean_temperature(
+            weather_readings)
+        mean_avg_humidity = self.calculate_mean_humidity(weather_readings)
+        print("Average Highest Temperature : " + str(int(avgmax_temp)) + "C")
```
Comment (ammadanwer) [inline fakhar-weatherman.py]: for concatenating strings, use `f"Average Highest Temperature : {str(avgmax_temp)}"`
or `"Average Highest Temperature {}".format(str(avgmax_temp))`

File: fakhar-weatherman.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+
+def search_files(path, arguments):
+    """ Read Files based on the year/month Mentioned """
+    weather_readings = []
+    # mentioned_date = arguments[0][0][1]
+    mentioned_date = arguments.split(
+        '/') if '/' in arguments else arguments  # Parsing MONTH AND YEAR if Provided
+    # Joining the Year Mentioned in File Naming Format
+    year = "".join(mentioned_date[0] + "_" + monthDictionary[int(mentioned_date[1])]
```
Comment (ammadanwer) [inline fakhar-weatherman.py]: format string here as well.

File: fakhar-weatherman.py (hunk 1 above)
Comment (efyudi) [reply fakhar-weatherman.py]: oh. simple mistake. I will make it correct

**Pull request merged**